internal/bminventory/inventory.go

```diff
@@ -936,12 +936,9 @@ func (b *bareMetalInventory) updateClusterData(ctx context.Context, cluster *com
 	if params.ClusterUpdateParams.VipDhcpAllocation != nil && swag.BoolValue(params.ClusterUpdateParams.VipDhcpAllocation) != vipDhcpAllocation {
 		updates["vip_dhcp_allocation"] = swag.BoolValue(params.ClusterUpdateParams.VipDhcpAllocation)
 		vipDhcpAllocation = swag.BoolValue(params.ClusterUpdateParams.VipDhcpAllocation)
```

Comment:
If you change the order of these 2 lines, than we can convert the VipDhcpAllocation to bool once and use it in the assignment to `updates["vip_dhcp_allocation"] =`